src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -121,21 +130,14 @@ export default class DefaultMessagingSession implements MessagingSession {
       }
     });
     this.setUpEventListeners();
+    this.isConnecting = false;
```

Comment:
If any error occurs in this method, `this.isConnecting` will always be `true`. For example, `getMessagingSessionEndpoint` throws an error if the customer passes the Chime client with expired credentials.

How about using try/catch/finally block?

```
try {
  await ...
} catch {
} finally {
  this.isConnecting = false;
}
```

Reply:
thanks, updated